### ISSUE TITLE
Modifying the columns number input kills the content

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#15525]
**Describe the bug**
When we alter the number of columns using the input instead of the slider the content is not saved.

**To reproduce**
1. In a post or page
2. Add a column block
3. Write some text in both blocks
4. Select the columns block
5. Modify the number of columns in the inspector _by changing the number in the input_, instead of the slider, e.g. instead of 2, delete 2 and write 3
6. Content is gone.

**Expected behavior**
To behave like the slider:

- If I am adding columns the content remains
- If I am removing the content is deleted RTL 
- If I add the same number the content is saved

**Screenshots**

![collumns-no-content](https://user-images.githubusercontent.com/107534/57438877-88855b00-724d-11e9-95b1-b2065a73b13d.gif)

**Comment by youknowriad** [comment]
This is related to #9009 while not exactly the same thing, I think they should have a unique solution. Would you mind adding details to that issue.

**Comment by draganescu** [comment]
@youknowriad I'll add there, but that problem (#9009) is about adding rows to columns, while this looks like a bug in handling that number input, a much smaller issue I'd say, right?

**Comment by youknowriad** [comment]
yes, smaller one but if the fix for the other issue is to avoid the "input" and use buttons to add/remove columns, this will be fixed automatically. I don't mind reopening this issue but it feels like the solution could fix both

**Comment by draganescu** [comment]
good point on removing the input!
